File: src/dataset/c/kindergarten_garden_Exercism_m3g4d1v3r/gemini/GeminiFlash_kindergarten_garden_v1.c
#include "kindergarten_garden.h"

#include <string.h>

static const char *students[] = {"Alice",  "Bob",    "Charlie", "David",
                                 "Eve",    "Fred",   "Ginny",   "Harriet",
                                 "Ileana", "Joseph", "Kincaid", "Larry"};

plants_t plants(const char *diagram, const char *student) {
    plants_t result = {{DIRT, DIRT, DIRT, DIRT}};

    if (diagram == NULL || student == NULL) return (result);

    size_t diagram_length = strlen(diagram);
    if (diagram_length == 0) return (result);

    size_t row_length = 0;
    while (row_length < diagram_length && diagram[row_length] != '\n') {
        row_length++;
    }

    if (row_length == 0 || row_length > (sizeof(students) / sizeof(students[0]) * 2)) return (result);

    int student_index = -1;
    for (size_t i = 0; i < sizeof(students) / sizeof(students[0]); i++) {
        if (strcmp(student, students[i]) == 0) {
            student_index = (int)i;
            break;
        }
    }

    if (student_index == -1) return (result);

    size_t student_offset = (size_t)student_index * 2;

    if (student_offset >= row_length || student_offset + row_length + 1 >= diagram_length) return (result);

    char plant_chars[4] = {
        diagram[student_offset],
        diagram[student_offset + 1],
        diagram[student_offset + row_length + 1],
        diagram[student_offset + row_length + 2]
    };

    for (int i = 0; i < 4; i++) {
        switch (plant_chars[i]) {
            case 'C':
                result.plants[i] = CLOVER;
                break;
            case 'G':
                result.plants[i] = GRASS;
                break;
            case 'R':
                result.plants[i] = RADISHES;
                break;
            case 'V':
                result.plants[i] = VIOLETS;
                break;
            default:
                result.plants[i] = DIRT; // Handle unexpected characters
                break;
        }
    }

    return (result);
}

File: src/dataset/typescript/kindergarten-garden_Exercism-typescript-FilipeCerejo/gemini/GeminiFlash_kindergarten-garden_v1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Garden = void 0;
const DEFAULT_STUDENTS = [
    'Alice',
    'Bob',
    'Charlie',
    'David',
    'Eve',
    'Fred',
    'Ginny',
    'Harriet',
    'Ileana',
    'Joseph',
    'Kincaid',
    'Larry',
];
const PLANT_CODES = {
    G: 'grass',
    V: 'violets',
    R: 'radishes',
    C: 'clover',
};
class Garden {
    constructor(diagram, students = DEFAULT_STUDENTS) {
        this._pots = diagram;
        this._students = [...students].sort(); // Create a copy to avoid modifying the original array
        this._studentIndex = new Map();
        for (let i = 0; i < this._students.length; i++) {
            this._studentIndex.set(this._students[i], i);
        }
    }
    plants(student) {
        const studentIndex = this._studentIndex.get(student);
        if (studentIndex === undefined) {
            return []; // Or throw an error, depending on desired behavior
        }
        const firstPlantIdx = studentIndex * 2;
        const plants = [];
        plants.push(PLANT_CODES[this._pots[firstPlantIdx + 0]]);
        plants.push(PLANT_CODES[this._pots[firstPlantIdx + 1]]);
        plants.push(PLANT_CODES[this._pots[this._pots.indexOf('\n') + firstPlantIdx + 0 + 1]]);
        plants.push(PLANT_CODES[this._pots[this._pots.indexOf('\n') + firstPlantIdx + 1 + 1]]);
        return plants;
    }
}
exports.Garden = Garden;
